enhancements/api-review/custom-tags-aws.md

```diff
@@ -73,71 +95,397 @@ Note existing unchanged behavior: The installer will apply these tags to all AWS
+  As there are limitations on the maximum number of user-defined tags that can be set on an AWS resource, the ability to delete user-defined tags enables user to replace and manage tags within limits allowed.
+
+- As a cluster administrator of OpenShift, I would like to be able to update user-defined tags managed by OpenShift for AWS resources. User will need to update the user-defined tags to different set of values during ongoing operations.
+  This can be supported by allowing edit on mutable `.spec.platformSpec.aws.resourceTags` field in Infrastructure CR.
+
+- As a cluster administrator of OpenShift, I would like to be able to remove user-defined tags from `.spec.platformSpec.aws.resourceTags` without deleting for AWS resources.
+  This enables, user-defined tags to be modified on AWS without being overridden by OpenShift operators. The user can adopt new 3rd party component to manage tags.
+
+- As a cluster administrator of OpenShift, I expect user-defined tags added in Infrastructure CR are reconciled and desired user-defined tags maintained on AWS resources.
+  The latest updates on `.spec.platformSpec.aws.resourceTags` field must be reconciled to AWS resources and desired user-defined tags maintained when being managed by OpenShift.
+  Any modifications which are external to OpenShift must be reconciled to desired user-defined tags from `.spec.platformSpec.aws.resourceTags`.
+
+The following user stories are added to support existing methods of updating user-defined tags which override `.spec.platformSpec.aws.resourceTags` to avoid breaking change.
+As the existing cluster might be configured with user-defined tags from kube resources, the precedence is given to user-defined tags in kube resources.
+- As a cluster administrator of OpenShift, I expect user-defined tags added in kube resources override the entries in `.spec.platformSpec.aws.resourceTags`.
+
+- As a cluster administrator of OpenShift, I expect user-defined tags added in kube resources must not be overridden by update or delete actions.
+
+
+
+### API Extensions
+
+This proposal edits `Infrastructure.config.openshift.io/v1` type
+
+The Infrastructure resource example to use spec for api changes
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services Infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+    // Existing fields
+    ...
+    // ResourceTags is a list of additional tags to apply to AWS resources created for the cluster.
+    // See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+    // AWS supports a maximum of 10 tags per resource. OpenShift reserves 5 tags for its use, leaving 5 tags
+    // available for the user.
+    // ResourceTags field is mutable and items can be removed.
+    // +kubebuilder:validation:MaxItems=10
```

Comment:
As OpenShift reserves 5, should this not be `MaxItems=5`?

Reply:
I do not know of any. OpenShift uses the kubernetes.io/cluster/... tag on all resources. Some operators may add other tags. The 5 reservations is for future use so that we could add more tags if needed.

Reply:
> the MaxItems includes both Openshift and user tags as a combined limit.

Can you explain why it includes both? At the moment, as a user, I could add 10 tags here and go over the limit of 5 that I'm allowed. I think what we want to do is limit this list to 5 items which allows the operators to add their own tags (eg the cluster tag and name tags) without going over the 10 tag limit.

Reply:
This still needs to be fixed, it should be 5 items